### ISSUE TITLE
Works with github actions

### ISSUE THREAD

**Issue by Kingmidas74** [sobolevn/git-secret#628]
What are the steps to reproduce this issue?
-------------------------------------------
1. Create steps in github actions yaml script
   `steps:

            - name: Checkout sources
              uses: actions/checkout@v2

            - name: Configure GPG Key
              uses: crazy-max/ghaction-import-gpg@v3
              with:
                gpg-private-key: ${{ secrets.GPG_SIGNING_KEY }}
                passphrase: ${{ secrets.SECRET_PWD }}
                git-user-signingkey: true
                git-commit-gpgsign: true                
            
            - name: Reveal secrets
              env:
                SECRET_PWD: ${{ secrets.SECRET_PWD }}
              run: |
                sudo apt install git-secret              
                sudo chmod -R 777 ./secrets
                git secret tell myuseremail@mail.com                
                git secret reveal -p ${{ secrets.SECRET_PWD }}`
2. Get an error 

> Process completed with exit code 1.

3. Well done!

What happens?
I'm trying to use git-secret with a GitHub Actions and receive error without any description 
…

What were you expecting to happen?
I think "secret" files will be decrypted.
…

Any logs, error output, etc?
----------------------------
Basically... That's all:
`done. myuseremail@mail.com added as someone who know(s) the secret.
cleaning up...
Error: Process completed with exit code 1.`

Any other comments?
my SO question: https://stackoverflow.com/questions/65385671/git-secret-reveal-failed-on-github-actions
I've enabled external logs for action with a special Github env secrets. Still no add info about error.
…

What versions of software are you using?
----------------------------------------
I don't know how to get this info from github actions. I install git-secret from apt.

**Comment by sobolevn** [comment]
Can you please try with the latest `master`?

**Comment by Kingmidas74** [comment]
Well.. I tried.
`##[debug]Loading env
Run git clone https://github.com/sobolevn/git-secret.git git-secret
  git clone https://github.com/sobolevn/git-secret.git git-secret
  cd git-secret && make build
  PREFIX="/usr/local" make install
  shell: /bin/bash -e {0}
##[debug]/bin/bash -e /home/runner/work/_temp/cfa80803-a90c-45fa-a9d1-61ba2566e2bb.sh
Error: No such file or directory`

**Comment by joshrabinowitz** [comment]
Can you retry with `SECRETS_VERBOSE=1` set (or, using git-secret's `-v` option where available)
it should give your more info about where the failure is occurring.

**Comment by Kingmidas74** [comment]
My full steps from yml file
`steps:

            - name: Install git-secret
              env: 
                SECRETS_VERBOSE: 1
                PREFIX: "/usr/local"
              run: |   
                git clone https://github.com/sobolevn/git-secret.git git-secret
                cd git-secret && make build
                make install

            - name: Configure GPG Key
              uses: crazy-max/ghaction-import-gpg@v3
              with:
                gpg-private-key: ${{ secrets.GPG_SIGNING_KEY }}
                passphrase: ${{ secrets.SECRET_PWD }}
                git-user-signingkey: true
                git-commit-gpgsign: true

            - name: Checkout sources
              uses: actions/checkout@v2
              
            - name: Reveal secrets
              env:
                SECRET_PWD: ${{ secrets.SECRET_PWD }}
                SECRETS_VERBOSE: 1              
              run: |                
                git secret tell 
                git secret reveal -p ${{ secrets.SECRET_PWD }}
            
            - name: Build images
              run: docker-compose build`

And github action logs on step 1:
`##[debug]Evaluating condition for step: 'Install git-secret'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: Install git-secret
##[debug]Loading inputs
##[debug]Loading env
Run git clone https://github.com/sobolevn/git-secret.git git-secret
##[debug]/bin/bash -e /home/runner/work/_temp/f8f78535-d469-4363-80d1-541c3c9e8795.sh
Cloning into 'git-secret'...
cat src/version.sh src/_utils/_git_secret_tools_freebsd.sh src/_utils/_git_secret_tools.sh src/_utils/_git_secret_tools_osx.sh src/_utils/_git_secret_tools_linux.sh src/commands/git_secret_list.sh src/commands/git_secret_tell.sh src/commands/git_secret_hide.sh src/commands/git_secret_changes.sh src/commands/git_secret_cat.sh src/commands/git_secret_killperson.sh src/commands/git_secret_clean.sh src/commands/git_secret_remove.sh src/commands/git_secret_whoknows.sh src/commands/git_secret_usage.sh src/commands/git_secret_init.sh src/commands/git_secret_add.sh src/commands/git_secret_reveal.sh src/main.sh > "git-secret"; \
chmod +x git-secret; sync
chmod +x "./utils/install.sh"; sync; \
"./utils/install.sh" "/usr/local"
cp: cannot create regular file '/usr/local/bin/git-secret': Permission denied
make: *** [install] Error 1
Makefile:25: recipe for target 'install' failed
Error: Process completed with exit code 2.
##[debug]Finishing: Install git-secret`

I don't see any sense in SECRETS_VERBOSE.  
I've tried `sudo chmod -R 777 /usr` but no success, more errors only.

**Comment by Kingmidas74** [comment]
Update. 
I've fixed setup error by added sudo.
Now same error (in original issue).
Steps:
`steps:

            - name: Install git-secret
              env: 
                SECRETS_VERBOSE: 1
                PREFIX: "/usr/local"
              run: |   
                git clone https://github.com/sobolevn/git-secret.git git-secret
                cd git-secret
                sudo make build
                sudo make install              

            - name: Checkout sources
              uses: actions/checkout@v2

            - name: Configure GPG Key
              uses: crazy-max/ghaction-import-gpg@v3
              with:
                gpg-private-key: ${{ secrets.GPG_SIGNING_KEY }}
                passphrase: ${{ secrets.SECRET_PWD }}
                git-user-signingkey: true
                git-commit-gpgsign: true
              
            - name: Reveal secrets
              env:
                SECRET_PWD: ${{ secrets.SECRET_PWD }}
                SECRETS_VERBOSE: 1              
              run: |                
                git secret tell -m
                git secret reveal -p ${{ secrets.SECRET_PWD }}`

logs:
`##[debug]/bin/bash -e /home/runner/work/_temp/d9fc4168-85cf-491b-ae20-64c4628257aa.sh
gpg: key 36AE1B86671EAE98: "Name (default gpg) <myemail@mail.com>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
git-secret: done. myemail@mail.com added as user(s) who know the secret.
Error: Process completed with exit code 1.`

**Comment by joshrabinowitz** [comment]
I'm not sure exactly the cause, but I think it's related to the output `gpg: unchanged: 1`. 
I was able to get `git secret tell email@example.com` to set `$?` to `1` the second time I ran it.
Maybe we should start explicitly calling `exit( $exit_code_here )` in `src/main.sh`.

**Comment by joshrabinowitz** [comment]
Also, 
* you don't need to call `git secret tell` at all in your steps; for `git secret reveal` to work, that user must already be in the repo's keychain (in repo/.gitsecret), and the secrets already be decryptable by that user's secret key.

**Comment by Kingmidas74** [comment]
Okay. 
I've removed `git secret tell` from steps and it works now! Thanks for your help.

By the way: I've wrote small action for install git-steps (from source) in job process. https://github.com/marketplace/actions/install-git-secret
